src/range-slider/RangeSlider.tsx

```diff
@@ -0,0 +1,67 @@
+// @ts-ignore
+import React, { InputHTMLAttributes, ReactNode, useState } from "react";
+
+import { VariantProps, cva } from "class-variance-authority";
+import { ChangeEvent, forwardRef } from "react";
+
+const rangeSlider = cva(
+  "appearance-none h-3 rounded-full bg-gray-300 accent-pink-500",
+  {
+    variants: {
+      size: {
+        small: "w-32",
+        medium: "w-64",
+        large: "w-96",
+        unbound: "w-full",
+      },
+    },
+    defaultVariants: {
+      size: "medium",
+    },
+  }
+);
+
+type Props = {
+  onVolumeChange: (volume: number) => void;
+  min?: number;
+  max?: number;
+  label?: ReactNode | ReactNode[];
+} & Omit<InputHTMLAttributes<HTMLInputElement>, "size"> &
+  VariantProps<typeof rangeSlider>;
+
+export const RangeSlider = forwardRef<HTMLInputElement, Props>(
+  (
+    {
+      onVolumeChange,
+      className,
+      size,
+      min = 0,
+      max = 100,
+      label = undefined,
+      ...rest
+    },
+    ref
+  ) => {
+    const handleVolumeChange = (event: ChangeEvent<HTMLInputElement>) => {
```

Comment:
I'm in a debate about whether there is any value in this forward ref. 
The reason for the inclusion is the case of using a state management lib such as `react-hook-form` could allow the state management lib to control the range state via a ref. 

I'm not precious about keeping it, might be removed in the future is if this is providing no value 



Tl;Dr might need forward ref in future, API is easy to use from the beginning than add onto an existing component, but is very easy to component imo